src/NodaTime.Demo/DateIntervalDemo.cs

```diff
@@ -0,0 +1,101 @@
+﻿using NUnit.Framework;
+
+namespace NodaTime.Demo
+{
+    public class DateIntervalDemo
+    {
+        [Test]
+        public void Construction()
+        {
+            var calendar = CalendarSystem.Gregorian;
+            LocalDate start = new LocalDate(2017, 1, 1, calendar);
+            LocalDate end = new LocalDate(2017, 12, 31, calendar);
+
+            DateInterval interval = Snippet.For(new DateInterval(start, end));
+
+            Assert.AreEqual(365, interval.Length);
+            Assert.AreEqual("[2017-01-01, 2017-12-31]", interval.ToString());
+            Assert.AreEqual(start, interval.Start);
+            Assert.AreEqual(end, interval.End);
+            Assert.AreEqual(calendar, interval.Calendar);
+        }
+
+        [Test]
+        public void Intersection()
```

Comment:
We can only handle a single snippet per method - I'd suggest that an intersection between "January-August" and "June-December" would be good.